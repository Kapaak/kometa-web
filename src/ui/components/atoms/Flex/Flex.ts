import { css, styled } from 'styled-components';

type justifyOptions =
  | 'space-between'
  | 'space-evenly'
  | 'center'
  | 'space-around'
  | 'end'
  | 'flex-end'
  | 'flex-start'
  | 'left'
  | 'right';

type alignOptions = 'center' | 'flex-end' | 'flex-start' | 'stretch';

type positionOptions = 'absolute' | 'relative' | 'fixed';

type directionOptions = 'row' | 'column' | 'row-reverse' | 'column-reverse';

type textAlignOptions = 'left' | 'center' | 'right';

type wrapOptions = 'nowrap' | 'wrap' | 'wrap-reverse';

type FlexProps = {
  justify?: justifyOptions;
  align?: alignOptions;
  direction?: directionOptions;
  gap?: string;
  textAlign?: textAlignOptions;
  width?: string;
  height?: string;
  position?: positionOptions;
  wrap?: wrapOptions;
  padding?: string;
  flex?: string;
  minWidth?: string;
  maxWidth?: string;
  order?: number;
};

export const Flex = styled.div.withConfig({
  shouldForwardProp: (prop) =>
    ![
      'justify',
      'align',
      'direction',
      'gap',
      'textAlign',
      'width',
      'height',
      'position',
      'wrap',
      'padding',
      'flex',
      'minWidth',
      'order',
    ].includes(prop),
})<FlexProps>`
  ${({
    position = 'relative',
    direction = 'row',
    justify = 'flex-start',
    align = 'stretch',
    gap = '0',
    textAlign = 'left',
    width = 'auto',
    height = 'auto',
    wrap = 'nowrap',
    padding = '0',
    flex = '0 1 auto',
    minWidth = 'auto',
    maxWidth = '100%',
    order = 'initial',
  }) => {
    return css`
      display: flex;
      position: ${position};
      padding: ${padding};
      flex-direction: ${direction};
      justify-content: ${justify};
      align-items: ${align};
      gap: ${gap};
      height: ${height};
      width: ${width};
      text-align: ${textAlign};
      flex-wrap: ${wrap};
      flex: ${flex};
      min-width: ${minWidth};
      max-width: ${maxWidth};
      order: ${order};
    `;
  }}
`;
