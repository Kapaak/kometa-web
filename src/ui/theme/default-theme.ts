import { DefaultTheme } from 'styled-components';

import { calculateEmFromPx } from '~/utils/dimensions';

export const theme: DefaultTheme = {
  colors: {
    primary: {
      light: '#DCE9FA',
      main: '#014B8B',
      dark: '#1A5FBF',
    },
    secondary: {
      light: '#FFCA0F',
      main: '#FFCA0F',
      dark: '#e6b812',
    },
    tetriary: {
      light: '#73F0C0',
      main: '#73F0C0',
      dark: '#73F0C0',
    },
    error: {
      light: '#FF0000',
      main: '#FF0000',
      dark: '#FF0000',
    },
    success: {
      light: '#34C759',
      main: '#34C759',
      dark: '#34C759',
    },
    grey: {
      100: '#fff',
      200: '#F8F8F8',
      400: '#EDEDED',
      500: '#DADADA',
      700: '#9A9A9A',
      900: '#000000',
    },
  },
  typography: {
    h1: {
      fontSize: '2.4rem',
      fontWeight: 600,
      lineHeight: '3.6rem',
    },
    h2: {
      fontSize: '5.6rem',
      fontWeight: 400,
      lineHeight: '4.8rem',
    },
    h3: {
      fontSize: '3rem',
      fontWeight: 400,
      lineHeight: '4.8rem',
    },
    body1: {
      fontSize: '1.8rem',
      fontWeight: 400,
      lineHeight: '2.4rem',
    },
    body2: {
      fontSize: '1.6rem',
      fontWeight: 400,
      lineHeight: '2.4rem',
    },
    body3: {
      fontSize: '1.6rem',
      fontWeight: 600,
      lineHeight: '3.2rem',
    },
    body4: {
      fontSize: '1.4rem',
      fontWeight: 400,
      lineHeight: '2.4rem',
    },
    body5: {
      fontSize: '1.4rem',
      fontWeight: 500,
      lineHeight: '2.4rem',
    },
    body6: {
      fontSize: '1.4rem',
      fontWeight: 600,
      lineHeight: '2.4rem',
    },
  },
  maxWidth: {
    small: '108rem',
    main: '145rem',
    wide: '145rem',
  },
  shadows: {
    main: '0px 0px 10px 0px rgba(0, 0, 0, 0.05)',
    dark: '0px 0px 15px 0px rgba(0, 0, 0, 0.15)',
  },
  fonts: {
    primary: '"Poppins", sans-serif',
    secondary: '"Bebas Neue", sans-serif',
  },
  breakpoints: {
    sm: calculateEmFromPx(600),
    md: calculateEmFromPx(790),
    lg: calculateEmFromPx(1024),
    xl: calculateEmFromPx(1200),
    xxl: calculateEmFromPx(1920),
  },
};
