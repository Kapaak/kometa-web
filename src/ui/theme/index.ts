export * from './default-theme';
export * from './global';
export * from './style-reset';
export * from './theme-wrapper';
