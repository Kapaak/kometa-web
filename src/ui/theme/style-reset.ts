import { css } from 'styled-components'

export const styleReset = css`
  *,
  *::before,
  *::after {
    margin: 0;
    padding: 0;
    padding: 0;
    border: none;
    font: inherit;
    text-transform: inherit;
    letter-spacing: inherit;
    color: inherit;
    appearance: none;
    border-radius: 0;
    background: none;
  }
`
