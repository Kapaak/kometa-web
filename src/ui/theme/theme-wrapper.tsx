import { ThemeProvider } from 'styled-components'

import { theme } from './default-theme'

export function withTheme(children: React.ReactElement) {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}
