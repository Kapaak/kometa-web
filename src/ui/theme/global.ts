import { createGlobalStyle, css } from 'styled-components';

import { styleReset } from './style-reset';

export const GlobalStyle = createGlobalStyle(
  ({ theme: { colors, fonts } }) => css`
    ${styleReset}

    *,
    *::before,
    *::after {
      box-sizing: border-box;
      font-family: ${fonts.primary};
    }

    html {
      font-size: 62.5%;
      min-height: 100%;
      height: 100%;
      scroll-behavior: smooth;
      box-sizing: inherit;
    }

    body {
      min-height: 100%;
      height: 100%;
      font: 400 1.6rem ${fonts.primary};
      text-rendering: optimizeLegibility;
      -webkit-font-smoothing: antialiased;
      -moz-osx-font-smoothing: grayscale;
      font-smooth: always;
      color: ${colors.grey['900']};
      background-color: ${colors.grey['200']};
    }

    //this class puts padding on body and it moves the whole page
    .yarl__no_scroll {
      padding-right: 0 !important;
    }

    br {
      //fixes br height in Firefox to be the same as in Chromium browsers
      margin-bottom: 2.3rem;
    }

    strong {
      font-weight: 600;
    }

    #__next {
      min-height: 100%;
      height: 100%;
      //To prevent the scale down if some item overflows
      /* overflow: auto; */
      //This broke pages and on next page it would retain scroll position
      //TODO: FIX
    }

    a,
    a[href^='tel'],
    a[href^='mail'] {
      color: inherit;
      text-decoration: none;
    }

    input:-webkit-autofill,
    input:-webkit-autofill:hover,
    input:-webkit-autofill:focus,
    input:-webkit-autofill:active {
      -webkit-box-shadow: 0 0 0 3rem ${colors.grey[100]} inset;
    }
  `
);
