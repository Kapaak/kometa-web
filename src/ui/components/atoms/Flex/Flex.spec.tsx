import { render } from '@testing-library/react'

import { withTheme } from '~/ui/theme'

import { Flex } from './Flex'

describe('Flex', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      withTheme(<Flex direction="row">This is inside flex</Flex>)
    )

    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })
})
