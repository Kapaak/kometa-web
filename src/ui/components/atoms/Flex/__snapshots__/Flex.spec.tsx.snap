// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex should render successfully 1`] = `
<body>
  <div>
    <div
      class="sc-a0ffddf9-0 netTT"
    >
      This is inside flex
    </div>
  </div>
</body>
`;
